### ISSUE TITLE
Make referenced Blazor package's static web assets available at wwwroot/_content during build

### ISSUE THREAD

**Issue by diegofrata** [dotnet/aspnetcore#24663]
I am building a Blazor app with [AntDesignBlazor](https://github.com/ant-design-blazor/ant-design-blazor) and the package ships with some static web assets such as CSS, JS and LESS files. These assets are copied fine to wwwroot/_content after publishing my app.

But during build and debugging, these assets are only found in the NuGet package cache, this is a major inconvenience because it requires a bunch of hacks to do anything with these assets, here's a couple of things I currently cannot do with Blazor libraries which are dead simple with node.js libraries:

- I cannot user LigerShark WebOptimizer to preprocess anything in the _content/ folder when running and debug my app, because the assets are not physically there;

- I cannot run Gulp tasks to process any of these assets during build as their path is unpredictable -- there's no easy way to guess the correct package version number as there might be multiple versions of the same package. Also, there's no easy way to pass in command line parameters to Gulp, which makes setting GeneratePathProperty="true" useless;

### Describe the solution you'd like
Copy static assets into wwwroot/_contents as part of the build, not just as part of the published bundle. 

### Additional context

Currently my approach to bundling AntDesign along my own LESS files has been to use manual MsBuild tasks, which are not very accessible to developers:

1. Copy the assets from AntDesign\staticwebassets to a predictable folder, which I named blazor_modules, sitting at the root of my project.

```
    <ItemGroup>
        <PackageReference Include="AntDesign" Version="0.3.0-nightly-2008060551" GeneratePathProperty="true" />
    </ItemGroup>

    <ItemGroup>
        <Content Remove="blazor_modules\**" />
        <None Remove="blazor_modules\**" />
    </ItemGroup>

    <Target Name="CopyAntDesignLess" BeforeTargets="Build">
        <ItemGroup>
            <_CopyItems Include="$(PkgAntDesign)\staticwebassets\**\*.less" />
        </ItemGroup>
        <Copy SourceFiles="@(_CopyItems)" DestinationFolder="$(ProjectDir)\blazor_modules\AntDesign\%(RecursiveDir)" SkipUnchangedFiles="true" OverwriteReadOnlyFiles="true" Retries="3" RetryDelayMilliseconds="300" />
    </Target>

```

2. Reference the assets in my style.less file like this, so I can use variables and mixins in my project:

```
@import '../../blazor_modules/AntDesign/less/ant-design-blazor.less';

@primary-color: @gray-10;
@layout-header-background: @gray-12;
@text-color: @gray-9;
```

3. Bringing it all together by running Gulp as part of the build:

```
    <Target Name="GulpBuild" BeforeTargets="Build" AfterTargets="CopyAntDesignLess">
        <Exec Command="npm install" Condition="!Exists('node_modules')" />
        <Exec Command="npm run gulp" />
    </Target>
```

It took me considerable time to come to this solution -- tried many other things in between. I think Blazor saves a lot of time by using file providers and serving static web assets straight from the NuGet cache, but this same feature becomes a painpoint once you want to consume these same assets during development / build time. There must be a better way than continually expanding these MsBuild tasks.

**Comment by javiercn** [comment]
@diegofrata thanks for contacting us.

This is expected and by design. We avoid copying the assets around since it has the potential to slow down the build significantly and interfere with other things like source control.

There is MSBuild extensibility in place within static web assets to achieve this in the appropriate way, you don't need to rely on the package as you are doing. Hooking on to ResolveStaticWebAssetsInputs offers the list of static web assets from packages, referenced projects and current projects.

**Comment by diegofrata** [comment]
@javiercn Thanks for replying! Hooking on to ResolveStaticWebAssetsInputs does help with the asset discovery issue but it leaves a few other issues, like being unable to put the files into place as part of the build:

1. After referencing the Blazor project into my server project, I get the error -- therefore I need to use a different path than what it's expected (hence why I used blazor_modules and put it outside wwwroot):

  Microsoft.NET.Sdk.Razor.StaticWebAssets.targets(191, 5): Conflicting assets with the same path '/_content/AntDesign/less/affix/style/entry.less' for content root paths '<redacted>'.

2. It is still left to developers to figure out these mechanisms which are not well documented!

I believe this scenario is common enough that needs to be addressed. I understand the concerns regarding build times, hence why I suggested this feature as either opt-in or opt-out. And, even in a big solution, this feature is very likely to be used only in the last project in the chain, the one that will be run/published. Finally, I'd imagine it's possible to mitigate build times by not copying resources from unmodified referenced packages or projects.

This is feature is critical if we ever want to see AntDesign, Bootstrap and the likes being distributed and used as NuGet packages. Is there any chance this issue can be raised more widely? I strongly believe it deserves a very serious consideration! :)
